src/Microsoft.AspNet.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -84,9 +84,11 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
 
             if (formatter == null)
             {
-                var unsupportedContentType = Resources.FormatUnsupportedContentType(
+                var message = Resources.FormatUnsupportedContentType(
                     bindingContext.OperationBindingContext.HttpContext.Request.ContentType);
-                bindingContext.ModelState.AddModelError(modelBindingKey, unsupportedContentType);
+
+                var exception = new UnsupportedContentTypeException(message);
+                bindingContext.ModelState.AddModelError(modelBindingKey, exception, bindingContext.ModelMetadata);
```

Comment:
1. We should not place `string`s from our resources directly into a `ModelError`. That leads to more framework-provided error messages that users can't override, something we've recently _fixed_. Extend `IModelBindingMessageProvider` and so on to allow overrides, get the message from somewhere else (that users can override e.g. a `DataAnnotations` attribute), or don't call the `string` overloads of `[Try]AddModelError()`.
2. If it's necessary to provide both a message and an `Exception`, just call `[Try]AddModelError()` twice with the same `key`.